assert_test.go

```diff
@@ -0,0 +1,83 @@
+package taskflow_test
+
+import (
+	"reflect"
+	"strings"
+	"testing"
+)
+
+func failedMessage(message ...string) string {
+	if len(message) == 0 {
+		return "Failed"
+	}
+	return strings.Join(message, ",")
+}
```

Comment:
I think we can simplify it and just have all functions with the following signature:
```
func assertTrue(t testing.TB, value bool, msg string) {
```

I think we have everywhere a human-readable assertion message. If not, then simply we can add a message. I consider it as a good practice (http://xunitpatterns.com/Assertion%20Message.html).